src/trajectory/Interpolated.cpp

```diff
@@ -100,6 +100,17 @@ void Interpolated::evaluateDerivative(double _t, int _derivative,
     _tangentVector.setZero();
   }
 
+  if(_t==mWaypoints[0].t)
```

Comment:
> Because it is only when _t is exactly the same with mWaypoints[0].t , we do something special.

I still don't really understand why we're checking for exact equality. What if, due to numerical precision, we end up getting `mWaypoints[0].t - eps`? :confused: 

> Also, @mkoval do you think we shall not support evaluate Derivative when _t < mWaypoints[0].t?

I think we should support this. We had a long discussion about this last year and decided that we would not implement hard bounds checks in Aikido arrays, unless the underlying implementation absolutely requires it. The issue is that numerical precision can result in an algorithm querying the result slightly outside of the nominal bounds.

This numerical imprecision is often unavoidable. It's often more useful to return a reasonable value, e.g. by extrapolating the last segment or extending the trajectory with a constant segment than to report an error. @psigen and I got burned by this repeatedly when implementing `or_parabolicsmoother` - as you can see by the variety of `epsilon`s. :sweat_smile: